build error when building HelloWorld example using platformio

I'm having a few build errors when compiling with platformio. The main file is more or less identical to the one in the example with minor changes for the pins.

### Build Error
```
Processing ardiuno-nano (platform: atmelavr; board: nanoatmega328; framework: arduino; lib_ldf_mode: deep+; build_type: debug; lib_deps: SPI, cbm80amiga/ST7789 AVR@^1.2.1, adafruit/Adafruit GFX Library@^1.12.0)
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
CONFIGURATION: https://docs.platformio.org/page/boards/atmelavr/nanoatmega328.html
PLATFORM: Atmel AVR (5.1.0) > Arduino Nano ATmega328
HARDWARE: ATMEGA328P 16MHz, 2KB RAM, 30KB Flash
DEBUG: Current (avr-stub) External (avr-stub, simavr)
PACKAGES: 
 - framework-arduino-avr @ 5.2.0 
 - toolchain-atmelavr @ 1.70300.191015 (7.3.0)
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ deep+, Compatibility ~ soft
Found 8 compatible libraries
Scanning dependencies...
Dependency Graph
|-- SPI @ 1.0 (License: Unknown, Path: /home/blockzilla/.platformio/packages/framework-arduino-avr/libraries/SPI)
|-- ST7789 AVR @ 1.2.1 (License: Unknown, Path: /home/blockzilla/CLionProjects/ardiuno-nano-uart-converter/.pio/libdeps/ardiuno-nano/ST7789 AVR)
|   |-- Adafruit GFX Library @ 1.12.0 (License: Unknown, Path: /home/blockzilla/CLionProjects/ardiuno-nano-uart-converter/.pio/libdeps/ardiuno-nano/Adafruit GFX Library)
|   |   |-- Adafruit BusIO @ 1.17.0 (License: Unknown, Path: /home/blockzilla/CLionProjects/ardiuno-nano-uart-converter/.pio/libdeps/ardiuno-nano/Adafruit BusIO)
|   |   |   |-- Wire @ 1.0 (License: Unknown, Path: /home/blockzilla/.platformio/packages/framework-arduino-avr/libraries/Wire)
|   |   |   |-- SPI @ 1.0 (License: Unknown, Path: /home/blockzilla/.platformio/packages/framework-arduino-avr/libraries/SPI)
|   |   |-- Wire @ 1.0 (License: Unknown, Path: /home/blockzilla/.platformio/packages/framework-arduino-avr/libraries/Wire)
|   |   |-- SPI @ 1.0 (License: Unknown, Path: /home/blockzilla/.platformio/packages/framework-arduino-avr/libraries/SPI)
|   |-- Adafruit BusIO @ 1.17.0 (License: Unknown, Path: /home/blockzilla/CLionProjects/ardiuno-nano-uart-converter/.pio/libdeps/ardiuno-nano/Adafruit BusIO)
|   |   |-- Wire @ 1.0 (License: Unknown, Path: /home/blockzilla/.platformio/packages/framework-arduino-avr/libraries/Wire)
|   |   |-- SPI @ 1.0 (License: Unknown, Path: /home/blockzilla/.platformio/packages/framework-arduino-avr/libraries/SPI)
|   |-- Wire @ 1.0 (License: Unknown, Path: /home/blockzilla/.platformio/packages/framework-arduino-avr/libraries/Wire)
|   |-- SPI @ 1.0 (License: Unknown, Path: /home/blockzilla/.platformio/packages/framework-arduino-avr/libraries/SPI)
|-- Adafruit GFX Library @ 1.12.0 (License: Unknown, Path: /home/blockzilla/CLionProjects/ardiuno-nano-uart-converter/.pio/libdeps/ardiuno-nano/Adafruit GFX Library)
|   |-- Adafruit BusIO @ 1.17.0 (License: Unknown, Path: /home/blockzilla/CLionProjects/ardiuno-nano-uart-converter/.pio/libdeps/ardiuno-nano/Adafruit BusIO)
|   |   |-- Wire @ 1.0 (License: Unknown, Path: /home/blockzilla/.platformio/packages/framework-arduino-avr/libraries/Wire)
|   |   |-- SPI @ 1.0 (License: Unknown, Path: /home/blockzilla/.platformio/packages/framework-arduino-avr/libraries/SPI)
|   |-- Wire @ 1.0 (License: Unknown, Path: /home/blockzilla/.platformio/packages/framework-arduino-avr/libraries/Wire)
|   |-- SPI @ 1.0 (License: Unknown, Path: /home/blockzilla/.platformio/packages/framework-arduino-avr/libraries/SPI)
|-- Adafruit BusIO @ 1.17.0 (License: Unknown, Path: /home/blockzilla/CLionProjects/ardiuno-nano-uart-converter/.pio/libdeps/ardiuno-nano/Adafruit BusIO)
|   |-- Wire @ 1.0 (License: Unknown, Path: /home/blockzilla/.platformio/packages/framework-arduino-avr/libraries/Wire)
|   |-- SPI @ 1.0 (License: Unknown, Path: /home/blockzilla/.platformio/packages/framework-arduino-avr/libraries/SPI)
|-- Wire @ 1.0 (License: Unknown, Path: /home/blockzilla/.platformio/packages/framework-arduino-avr/libraries/Wire)
Building in debug mode
avr-g++ -o .pio/build/ardiuno-nano/firmware.elf -mmcu=atmega328p -Wl,--gc-sections -flto -fuse-linker-plugin -Og -g2 -ggdb2 .pio/build/ardiuno-nano/src/main.cpp.o -L.pio/build/ardiuno-nano -Wl,--start-group .pio/build/ardiuno-nano/lib8e9/libSPI.a .pio/build/ardiuno-nano/lib9e9/libWire.a ".pio/build/ardiuno-nano/lib111/libAdafruit BusIO.a" ".pio/build/ardiuno-nano/liba11/libAdafruit GFX Library.a" ".pio/build/ardiuno-nano/lib6a3/libST7789 AVR.a" .pio/build/ardiuno-nano/libFrameworkArduinoVariant.a .pio/build/ardiuno-nano/libFrameworkArduino.a -lm -Wl,--end-group
/tmp/ccsyy5sy.s: Assembler messages:
/tmp/ccsyy5sy.s:2552: Error: symbol `next' is already defined
/tmp/ccsyy5sy.s:2688: Error: symbol `next' is already defined
/tmp/ccsyy5sy.s:2854: Error: symbol `next' is already defined
lto-wrapper: fatal error: avr-g++ returned 1 exit status
compilation terminated.
/home/blockzilla/.platformio/packages/toolchain-atmelavr/bin/../lib/gcc/avr/7.3.0/../../../../avr/bin/ld: error: lto-wrapper failed
collect2: error: ld returned 1 exit status
*** [.pio/build/ardiuno-nano/firmware.elf] Error 1
======================================================================================================== [FAILED] Took 0.57 seconds ========================================================================================================
```

`platformio.ini`
```ini
[common]
monitor_speed = 115200

[env:ardiuno-nano]
platform = atmelavr
board = nanoatmega328
framework = arduino
lib_ldf_mode = deep+
build_type = debug
lib_deps =
    SPI
    cbm80amiga/ST7789 AVR@^1.2.1
    adafruit/Adafruit GFX Library@^1.12.0
```

`main.cpp`
```cpp
// ST7789 library example
// (c) 2019 Pawel A. Hernik

/*
 #01 GND -> GND
 #02 VCC -> VCC (3.3V only!)
 #03 SCL -> D13/SCK
 #04 SDA -> D11/MOSI
 #05 RES -> optional
 #06 DC  -> D10 or any digital
 #07 CS  -> D9 or any digital
*/

#include <SPI.h>
#include <Adafruit_GFX.h>
#include "ST7789_AVR.h"

#define TFT_DC 11
#define TFT_CS 12  // with CS
#define TFT_RST -1  // with CS
// #define TFT_CS -1 // without CS
// #define TFT_RST  9 // without CS

#define SCR_WD 240
#define SCR_HT 320
ST7789_AVR lcd = ST7789_AVR(TFT_DC, TFT_RST, TFT_CS);

void setup() {
 Serial.begin(115200);
 lcd.init(SCR_WD, SCR_HT);
 lcd.fillScreen(RED);
 lcd.setCursor(0, 0);
 lcd.setTextColor(WHITE, BLUE);
 lcd.setTextSize(3);
 lcd.println("HELLO WORLD"); // using Adafruit default font
}

void loop() {
}
```

--- a/.github/workflows/examples.yml
+++ b/.github/workflows/examples.yml
@@ -1,0 +1,33 @@
+name: PlatformIO Examples CI
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        example: [examples/ST7789_lib_HelloWorld]
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cache/pip
+            ~/.platformio/.cache
+          key: ${{ runner.os }}-pio
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.11'
+      - name: Install PlatformIO Core
+        run: pip install --upgrade platformio
+
+      - name: Build PlatformIO examples (release mode)
+        run: pio ci --lib="." -O "lib_deps=adafruit/Adafruit GFX Library@^1.12.0" --board=nanoatmega328new --board=uno
+        env:
+          PLATFORMIO_CI_SRC: ${{ matrix.example }}
+      - name: Build PlatformIO examples (debug mode)
+        run: pio ci --lib="." -O "build_type=debug" -O "lib_deps=adafruit/Adafruit GFX Library@^1.12.0" --board=nanoatmega328new --board=uno
+        env:
+          PLATFORMIO_CI_SRC: ${{ matrix.example }}

--- a/ST7789_AVR.cpp
+++ b/ST7789_AVR.cpp
@@ -193,7 +193,7 @@ inline void ST7789_AVR::copyMulti(uint8_t *img, uint16_t num)
 #ifdef COMPATIBILITY_MODE
   while(num--) { SPI.transfer(*(img+1)); SPI.transfer(*(img+0)); img+=2; }
 #else
-  uint8_t lo,hi;
+  uint8_t lo = 0, hi = 0;
   asm volatile
   (
   "2:\n"

--- a/ST7789_AVR.cpp
+++ b/ST7789_AVR.cpp
@@ -160,7 +160,7 @@ inline void ST7789_AVR::writeMulti(uint16_t color, uint16_t num)
 #else
   asm volatile
   (
-  "next:\n"
+  "1:\n"
     "out %[spdr],%[hi]\n"
     "rjmp .+0\n"  // wait 8*2+1 = 17 cycles
     "rjmp .+0\n"
@@ -180,7 +180,7 @@ inline void ST7789_AVR::writeMulti(uint16_t color, uint16_t num)
     "rjmp .+0\n"
     "nop\n"
     "sbiw %[num],1\n"
-    "brne next\n"
+    "brne 1b\n"
     : [num] "+w" (num)
     : [spdr] "I" (_SFR_IO_ADDR(SPDR)), [lo] "r" ((uint8_t)color), [hi] "r" ((uint8_t)(color>>8))
   );
@@ -196,7 +196,7 @@ inline void ST7789_AVR::copyMulti(uint8_t *img, uint16_t num)
   uint8_t lo,hi;
   asm volatile
   (
-  "nextCopy:\n"
+  "2:\n"
     "ld  %[hi],%a[img]+\n"
     "ld  %[lo],%a[img]+\n"
     "out %[spdr],%[lo]\n"
@@ -216,7 +216,7 @@ inline void ST7789_AVR::copyMulti(uint8_t *img, uint16_t num)
     "rjmp .+0\n"
     "nop\n"
     "sbiw %[num],1\n"
-    "brne nextCopy\n"
+    "brne 2b\n"
     : [num] "+w" (num)
     : [spdr] "I" (_SFR_IO_ADDR(SPDR)), [img] "e" (img), [lo] "r" (lo), [hi] "r" (hi)
   );

